zookeeper-server/src/main/java/org/apache/zookeeper/server/NettyServerCnxnFactory.java

```diff
@@ -775,15 +775,18 @@ public InetSocketAddress getLocalAddress() {
 
     private void addCnxn(final NettyServerCnxn cnxn) {
         cnxns.add(cnxn);
-        InetAddress addr = ((InetSocketAddress) cnxn.getChannel().remoteAddress()).getAddress();
-
-        ipMap.compute(addr, (a, cnxnCount) -> {
-            if (cnxnCount == null) {
-                cnxnCount = new AtomicInteger();
-            }
-            cnxnCount.incrementAndGet();
-            return cnxnCount;
-        });
+        increaseConnectionCreateCount();
+        synchronized (ipMap) {
```

Comment:
why do we need synchronisation in this case? ipMap is ConcurrentHashMap.
Also, this synchronization is inconsistent. we are not synchronizing in removeCnxnFromIpMap.